src/cmds/scripts/pbs_habitat.in

```diff
@@ -242,9 +242,9 @@ upgrade_pbs_database() {
 	sys_pgsql_ver=$(echo `${PGSQL_DIR}/bin/psql -V` | awk '{print $NF}' | cut -d '.' -f 1,2)
 	ver=`cat ${data_dir}/PG_VERSION`
 
-	result=1
-	[[ $ver -ge $sys_pgsql_ver ]] && result=0
-	if [ ${result} -eq 1 ]; then
+	[ ${ver%.*} -eq ${sys_pgsql_ver%.*} ] && [ ${ver#*.} \> ${sys_pgsql_ver#*.} ] || [ ${ver%.*} -gt ${sys_pgsql_ver%.*} ];
+	result=$?
+	if [ ${result} -eq 0 ]; then
```

Comment:
You can pick a more meaningful variable name such as `old_pgsql_ver` instead of `ver`. You should upgrade when `sys_pgsql_ver > old_pgsql_ver`. In this case your result will be 1 so upgrade routine is not getting invoked.